src/se.rs

```diff
@@ -1,94 +1,64 @@
 //! CBOR serialisation tooling
-use std::io::Write;
+use alloc::string::String;
+use alloc::vec::Vec;
+use core::convert::TryInto;
+use core::fmt::Write;
```

Comment:
No this is not the same `Write` you cannot do that. `std::io::Write` and `std::fmt::Write` are different trait for different purposes. Replacing the IO with fmt won't work. You need to feature gate with the trait instead.

You can read about how to use feature flags in https://doc.rust-lang.org/cargo/reference/features.html